examples/abitest/client/cpp/httptest.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef EXAMPLES_ABITEST_CLIENT_HTTPTEST_H
+#define EXAMPLES_ABITEST_CLIENT_HTTPTEST_H
+#define CPPHTTPLIB_OPENSSL_SUPPORT 1
```

Comment:
This `#define` is quite separated from where it has an effect &ndash; I think it would be clearer to move it into httptest.cc on the line before `#include "httplib.h"` happens.

Reply:
Ah, I think that's because `"httplib.h"` is also included in abiclient.cc (which in turn means that the two .cc files build two incompatible versions of the HTTP library into their .o files).

I think this is likely to confuse someone in future, though, so could we either:
 - add a comment here explaining things, or
 - add a local wrapper header file for `httplib.h` (maybe `httplib_config.h`?) and always include the wrapper
```C++
// Wrapper include file for httplib.h with all configured options in one place
// (so builds are consistent).
#define CPPHTTPLIB_OPENSSL_SUPPORT 1
#include "httplib.h"
```

(The latter is a bit more work as it will probably need a new Bazel target, but is probably the nicer option)